core/eth/validator_quorum_lookup.go

```diff
@@ -0,0 +1,124 @@
+package eth
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"math/big"
+
+	regcoordinator "github.com/Layr-Labs/eigenda/contracts/bindings/EigenDARegistryCoordinator"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	lru "github.com/hashicorp/golang-lru/v2"
+)
+
+// A utility for looking up which quorums a given validator is a member of at a specific reference block number.
+type ValidatorQuorumLookup interface {
+	// Get the list of quorums that the given validator is a member of, at the specified reference block number.
+	GetQuorumsForValidator(
+		ctx context.Context,
+		validatorAddress core.OperatorID,
+		referenceBlockNumber uint64) ([]core.QuorumID, error)
+}
+
+var _ ValidatorQuorumLookup = (*validatorQuorumLookup)(nil)
+
+// A standard implementation of the ValidatorQuorumLookup interface.
+type validatorQuorumLookup struct {
+	registryCoordinator *regcoordinator.ContractEigenDARegistryCoordinator
+}
+
+// Create a new ValidatorQuorumLookup instance.
+func NewValidatorQuorumLookup(
+	backend bind.ContractBackend,
+	registryCoordinatorAddress gethcommon.Address,
+) (ValidatorQuorumLookup, error) {
+
+	return nil, nil
+}
+
+func (v *validatorQuorumLookup) GetQuorumsForValidator(
+	ctx context.Context,
+	validatorID core.OperatorID,
+	referenceBlockNumber uint64,
+) ([]core.QuorumID, error) {
+
+	blockNumber := big.NewInt(int64(referenceBlockNumber))
+
+	opts := &bind.CallOpts{
+		Context:     ctx,
+		BlockNumber: blockNumber,
+	}
+
+	// This method returns a bitmap as a big.Int, where the first 255 bits represent membership in quorums 0-254.
+	bigIntBitmap, err := v.registryCoordinator.GetCurrentQuorumBitmap(opts, validatorID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get quorum bitmap: %w", err)
+	}
+	bitmap := bigIntBitmap.Bytes()
+
+	quorumIDs := make([]core.QuorumID, 0)
+
+	// Although technically 254 is the max quorum ID (due to an embarrassing off-by-one typo), it doesn't hurt
```

Comment:
I thought we determined that the max was something like 192 or something like that?